libstb/secvar/backend/edk2-compat-process.c

```diff
@@ -596,6 +627,11 @@ int process_update(struct secvar_node *update, char **newesl,
 	}
 	memcpy(*newesl, update->var->data + auth_buffer_size, *new_data_size);
 
+	/* Validate the new ESL is in right format */
+	rc = validate_esl_list(update->var->key, *newesl, *new_data_size);
```

Comment:
Candidate for calling the full `.validate()` function here, and including `validate_esl_list()` in there

Reply:
I thought edk2-compat-process.c is sort of internal file which hooks will call. And hooks to be called by secvar_main or platform thing.. I will not prefer edk2-compat-process calling hooks internally. I think it is .validate() function that should call validate_esl_list ().

Also, validate() will be doing much more.. like we can add a function validate_auth().. that does auth command validation.. So for certain I don't want to call validate() inside edk2-compat-process.c process_update() function

Reply:
Why would we not want to reuse code? If its doing much more, maybe we should call the validate function before we attempt to process the update at all, to ensure it works.

The hook doesn't need to be called from the driver struct, but I do think we should at least reuse either the function exposed in the struct, or the functions it calls. I'll take a look at this when I do a full file review

Reply:
Exactly we need to reuse the code.. and for that reason I think. .validate() hook should call validate_esl_list().  I don't want to mix up the interfaces. Well,  I am not sure if that explains the reason so well..but I still do not feel calling .validate() hook in edk2-compat-process.c  So.. sure go ahead and review it.. and if you still want to do it... its ok.. just that it is my choice not to do so..